### ISSUE TITLE
Scoped service act as transient after upgrade from 9.0.4 to 10.0.1

### ISSUE THREAD

**Issue by Elywejnak** [ChilliCream/hotchocolate#1025]
**Describe the bug**
Hello, we are using scoped service which is updated by ASP NET CORE (2.2) middleware and then is expected in various places in resolvers (and below). In 9.0.4 it acts as expected, service is created upon first dependency request, updated by middleware and then the same instance is distributed to all places (standard dependency injection - constructior / [Service] attribute in resolvers). After upgrade to 10.0.1 (latest in nuget) new instance is created, updated by middleware, but then for each dependency request in resolvers and nested services is new instance created. It happens only in graphql layer, aspnet middlewares are receiving the same instance as expected.

For a bit more context, it is Database service which holds information about master and tenant database connection. Tenant database connection is resolved in the middleware and then the initialized instance is expected in data layer (which is called from resolvers)

**To Reproduce**
I might try to put together smaller reproduction project, or rather test case, but depends if it is not for some reason expected behaviour. Didn't notice anything in docs.

**Expected behavior**
Scoped services in ASP.NET layer stay scoped in request for whole application

**ASP.NET CORE 2.2 + Hotchocolate 10.0.1 - Mac**

**Comment by michaelstaib** [comment]
Hi, 

a simple repro would be great. Although we changed some things around dependency injection we have not touched how we pull in services. And there should be no change. So, this might be a bug. 

Do you think you can provide a repro?

**Comment by Elywejnak** [comment]
[ScopedServiceIssue.zip](https://github.com/ChilliCream/hotchocolate/files/3531069/ScopedServiceIssue.zip)

Hi, here is simple (freshly created project from Hotchocolate template) repro. There is added IteratorService which holds state (one static and one local property) and is set as scoped dependency and middleware IteratorMiddleware which on each request iterates state of the service by 1. Then there is simple query shema returning the service state. 
`query {
  iterator,
  staticIterator
}`

Basically 'iterator' value should return 1. 'staticIterator' just iterates static property (just check that the middleware is doing something)

I've tried also 9.0.4 and it worked as expected.

Thanks for looking into this, I might try to take a deeper look over weekend.

**Comment by michaelstaib** [comment]
@Elywejnak awesome. I will look at this today.

**Comment by michaelstaib** [comment]
@Elywejnak I have looked into this and can verify your issue. We have added it to our 10.1.0 backlog and will fix it soon.

**Comment by qina83** [comment]
Is there a road map for 10.1.0 delivery?

**Comment by michaelstaib** [comment]
@qina83 end of next week

**Comment by qina83** [comment]
@michaelstaib thx

**Comment by michaelstaib** [comment]
We found the issue. The problem seems to be the way we populate the execution services. I think we will have a fix for this tomorrow as a preview.

**Comment by michaelstaib** [comment]
https://github.com/ChilliCream/hotchocolate/releases/tag/10.1.0-preview.8
